memguard.go

```diff
@@ -38,13 +39,21 @@ Similarly, if a function is given a LockedBuffer that has been
 destroyed, the call will return an ErrDestroyed.
 */
 type LockedBuffer struct {
+	*lockedBuffer
+	*finalizerGuard
+}
+
+// LockedBuffer is actually just a handle to lockedBuffer
+type lockedBuffer struct {
 	sync.Mutex
 	Buffer []byte
 
 	readOnly  bool
 	destroyed bool
 }
 
+type finalizerGuard [16]byte
```

Comment:
Is there any particular reason that the finalizerGuard is the size and type it is?